app/src/main/java/com/github/se/eventradar/viewmodel/MyQrCodeViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.github.se.eventradar.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class MyQrCodeViewModel @Inject constructor(private val userRepository: IUserRepository) :
+    ViewModel() {
+  private val _uiState = MutableStateFlow(MyQrCodeUiState())
+  val uiState: StateFlow<MyQrCodeUiState> = _uiState
+  val qrCodesFolder = "QR_Codes"
+
+  fun getUsername() {
+    viewModelScope.launch {
+      when (val userIdResource = userRepository.getCurrentUserId()) {
+        is Resource.Success -> {
+          val userId = userIdResource.data
+          // Now fetch the user data using the fetched user ID
+          when (val userResult = userRepository.getUser(userId)) {
+            is Resource.Success -> {
+              _uiState.value = _uiState.value.copy(username = userResult.data!!.username)
+            }
+            is Resource.Failure -> {
+              Log.d("MyQrCodeViewModel", "Error fetching username: ${userResult.throwable.message}")
+            }
+          }
+        }
+        is Resource.Failure -> {
+          Log.d("MyQrCodeViewModel", "Error fetching user ID: ${userIdResource.throwable.message}")
+        }
+      }
+    }
+  }
+
+  fun getQRCodeLink() {
+    viewModelScope.launch {
+      when (val userIdResource = userRepository.getCurrentUserId()) {
```

Comment:
If you combine this function with the one above, you won't have to fetch the userId multiple times. Moreover, if you save the id in the uiState, you can access it wherever you want.